src/features/courses/course-controller.ts

```diff
@@ -2224,6 +2251,49 @@ class CourseController {
         });
     }
 
+    async getGradeForQuestion(options: GetGradeForQuestionOptions): Promise<StudentGrade> {
+        const {
+            questionId,
+            userId,
+            includeWorkbooks
+        } = options;
+        const include: sequelize.IncludeOptions[] = [{
+            model: StudentGradeInstance,
+            as: 'gradeInstances',
+            required: false,
+            where: {
+                active: true,
+            }
+        }];
+        if (includeWorkbooks) {
+            include.push({
+                model: StudentWorkbook,
+                as: 'workbooks',
+                required: false,
+                where: {
+                    active: true
+                }
+            });
+        }
+        try {
+            const grade = await StudentGrade.findOne({
+                where: {
+                    courseWWTopicQuestionId: questionId,
+                    userId,
+                    active: true,
+                },
+                include
+            });
+            if (!_.isNil(grade)) {
+                return grade;
+            } else {
+                throw new Error('Did not find a student grade.');
+            }
+        } catch (e) {
+            throw new RederlyError(`Unable to get grade for user #${userId} and question #${questionId}`);
```

Comment:
Why are we try catching and then throwing again? Unless there is a specific error to send it makes sense to let it boil up so it can be handled accordingly

Reply:
Does it make sense to throw a wrapped error and include it as a parameter?